### PR TITLE
feat(controllers/tekton): add finalizers to controllers of Pipeline and PipelineRun in version v2alpha1

### PR DESCRIPTION
**Pull request by flamywhale**
support serial deletion of Tekton resources when deleting our Pipeline and PipelineRun resources

### CODE REVIEW

**Comment by flamywhale** [comment]
Lint test may fail because of not syncronizing with the master branch.

File: controllers/tekton/pipeline/pipeline_controller.go [hunk 1]
```diff
@@ -53,16 +57,54 @@ type Reconciler struct {
 func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 
-	klog.Infof("req: %v", req)
+	// for debug
+	// klog.Infof("req: %v", req)
```
Comment (LinuxSuRen) [inline controllers/tekton/pipeline/pipeline_controller.go:61]: Please remove these debug statements.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]

File: controllers/tekton/pipeline/pipeline_controller.go (hunk 1 above)
Comment (flamywhale) [reply controllers/tekton/pipeline/pipeline_controller.go]: Okay, I will push a commit to remove it.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm

File: controllers/tekton/pipeline/pipeline_controller.go [hunk 2]
```diff
@@ -78,21 +117,83 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
...
+}
+
 // reconcileTektonCrd transforms our Pipeline CRD to Tekton CRDs
 func (r *Reconciler) reconcileTektonCrd(ctx context.Context, namespace string, pipeline *devopsv2alpha1.Pipeline) error {
 	// print the pipeline name and the number of its tasks.
 	klog.Infof("Pipeline name: %s\tTask num: %d", pipeline.Name, len(pipeline.Spec.Tasks))
 
 	// transform tasks included in the pipeline to Tekton Tasks
-	klog.Info("Going to transform tasks included in the pipeline to Tekton Tasks.")
+	// klog.Info("Going to transform tasks included in the pipeline to Tekton Tasks.")
```
Comment (JohnNiang) [inline controllers/tekton/pipeline/pipeline_controller.go:188]: Please remove this useless comment.

File: controllers/tekton/pipeline/pipeline_controller.go [hunk 3]
```diff
@@ -78,21 +117,83 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
...
 	for _, task := range pipeline.Spec.Tasks {
 		if err := r.reconcileTektonTask(ctx, namespace, &task, pipeline.Name); err != nil {
 			klog.Error(err, "Failed to reconcile tekton task resources.")
 			return err
 		}
 	}
 
-	klog.Info("Going to transform Pipeline CRD to Tekton Pipeline CRD.")
+	// klog.Info("Going to transform Pipeline CRD to Tekton Pipeline CRD.")
```
Comment (JohnNiang) [inline controllers/tekton/pipeline/pipeline_controller.go:196]: Please remove this useless comment

Comment (flamywhale) [reply controllers/tekton/pipeline/pipeline_controller.go:196]: Okay.

File: controllers/tekton/pipeline/pipeline_controller.go (hunk 2 above)
Comment (flamywhale) [reply controllers/tekton/pipeline/pipeline_controller.go:188]: Okay.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm

**Pull request merged**